File: scm-java/src/main/scheme/primitives/PrimitiveCharLowerCaseP.java
package scheme.primitives;

import scheme.Primitive;
import scheme.Value;
import scheme.SourcePos;

public class PrimitiveCharLowerCaseP extends Primitive {
    @Override
    public String name() {
        return "char-lower-case?";
    }

    @Override
    public String info() {
        return "Syntax: (char-lower-case? char)\n" +
               "Library: (scheme char)\n" +
               "Description: Returns #t if char is a lowercase character.\n" +
               "Example:\n" +
               "  (char-lower-case? #\\a) => #t\n" +
               "  (char-lower-case? #\\A) => #f";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        return Character.isLowerCase(Value.asChar(arguments[0]));
    }
}

File: scm-csharp/scheme/primitives/PrimitiveCharLowerCaseP.cs
namespace scheme;

public class PrimitiveCharLowerCaseP : Primitive
{
    public override string Name()
    {
        return "char-lower-case?";
    }

    public override string Info()
    {
        return
            "Syntax: (char-lower-case? char)\n" +
            "Library: (scheme char)\n" +
            "Description: Returns #t if char is a lowercase character.\n" +
            "Example:\n" +
            "  (char-lower-case? #\\a) => #t\n" +
            "  (char-lower-case? #\\A) => #f";
    }
    
    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        return Char.IsLower(Value.AsChar(arguments[0]));
    }
}
